packages/vaadin-overlay/test/restore-focus.test.js

```diff
@@ -83,4 +83,30 @@ describe('restore focus', () => {
     await close(overlay);
     expect(overlay._getActiveElement()).to.equal(focusable);
   });
+
+  describe('restoreFocusNode', () => {
+    beforeEach(() => {
+      overlay.restoreFocusNode = focusInput;
+    });
+
+    it('should not restore focus on close by default (restore-focus-on-close=false)', async () => {
+      overlay.restoreFocusOnClose = false;
+      focusInput.focus();
+      await open(overlay);
+      // emulate focus leaving the input
+      focusInput.blur();
+      document.body.focus();
+      await close(overlay);
+      expect(overlay._getActiveElement()).to.not.equal(focusInput);
+    });
+
+    it('should restore focus to the restoreFocusNode', async () => {
+      overlay.restoreFocusOnClose = true;
+      focusable.focus();
+      await open(overlay);
+      focusable.blur();
```

Comment:
Is `blur` needed here?

Reply:
It is needed because `<vaadin-overlay>` will [only try to restore focus](https://github.com/vaadin/web-components/blob/master/packages/vaadin-overlay/src/vaadin-overlay.js#L671) in case the focus is inside the overlay or at the document element on close.

Reply:
> in case the focus is inside the overlay

Hmm, isn't the `focusable` element inside the overlay and therefore the overlay should restore focus?

Reply:
The `focusable` element is outside the overlay

Reply:
Oh, I'm sorry, I confused it with another "focusable" element that is above:

https://github.com/vaadin/web-components/blob/7ccdd6071753be5d2a1cfd466a662350e8b65614/packages/vaadin-overlay/test/restore-focus.test.js#L14-L16